src/gpt_resolve/pdf_generator.py

```diff
@@ -59,6 +86,22 @@ def generate_solutions_pdf(
         solutions_dir.glob("*_solution.txt"),
         key=lambda x: int(x.stem.split("_")[0][1:]),
     )
+        # Get all solution files sorted
+    solution_files = sorted(
+        solutions_dir.glob("*_solution.txt"),
+        key=lambda x: int(x.stem.split("_")[0][1:])
+    )
+
+    # Validate each solution before adding to the document
+    for sol_file in solution_files:
+        print(f"Validating solution: {sol_file.name}")
+        content = sol_file.read_text(encoding="utf-8")
+        
+        if not compile_solution(content):
+            print(f"Error: Solution '{sol_file.name}' has LaTeX syntax errors. Please fix it.")
+            return 
+
+    print("All solutions are valid. Generating the final PDF...")
```

Comment:
Nice! But I think there is a more concise way to solve this:
- instead of looping through questions twice, you can use the existent loop through solution_files to, for each content read, try to compile it with compile_solution in a tmp file as you've suggested
- if the compilation is successful, continue the loop as you've done

Besides, with this implementation the user will only see the first error found. It'd be better if you collect wrong files in a list so that we print all files with errors at the end of the loop.